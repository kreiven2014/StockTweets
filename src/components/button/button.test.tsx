/* REACT */
import React from "react";

/* MODULES */
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

/* CUSTOM MODULES */
import Button from "./button";

describe("Button tests", () => {
  const mockHandler = jest.fn(() => {});

  test("renders correctly", (): void => {
    const tree = renderer
      .create(
        <Button
          text="forgot_password:btn_reset_password"
          onPress={mockHandler}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders correctly with child", (): void => {
    const tree = renderer
      .create(<Button onPress={mockHandler}>Test button text</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders correctly when disabled and reversed", (): void => {
    const tree = renderer
      .create(
        <Button
          disabled
          reversed
          text="forgot_password:btn_reset_password"
          // FUNCTIONS
          onPress={jest.fn(mockHandler)}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders content as null when no children and no text are provided", (): void => {
    const tree = renderer
      .create(
        <Button
          // FUNCTIONS
          onPress={jest.fn(mockHandler)}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("check on press handler", (): void => {
    const component = shallow(
      <Button
        text="forgot_password:btn_reset_password"
        onPress={mockHandler}
      />,
    );
    // TODO: adapter don't return tree of react-native elements.
    // should return TouchableOpacity
    const button = component
      .find({
        testID: "button",
      })
      .first();
    button.simulate("press");
    expect(component.prop("onPress")).toHaveBeenCalled();
  });

  // TODO(anybody): for some reason even if button is disabled onPress handler is still called !
  // test("check on press handler on disabled button", (): void => {
  //   const component = shallow(
  //     <Button
  //       disabled
  //       text="forgot_password:btn_reset_password"
  //       onPress={mockHandler}
  //     />
  //   );
  //   const button = component.find("TouchableOpacity").first();
  //   button.simulate("press");
  //   expect(component.prop("onPress")).not.toHaveBeenCalled();
  // });
});
