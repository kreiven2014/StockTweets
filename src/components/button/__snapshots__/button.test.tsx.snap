// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests renders content as null when no children and no text are provided 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#676B7A",
      "borderRadius": 4,
      "height": 68.5,
      "justifyContent": "center",
      "opacity": 1,
      "width": "100%",
    }
  }
  testID="button"
/>
`;

exports[`Button tests renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#676B7A",
      "borderRadius": 4,
      "height": 68.5,
      "justifyContent": "center",
      "opacity": 1,
      "width": "100%",
    }
  }
  testID="button"
>
  <Text
    allowFontScaling={false}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 24,
          "letterSpacing": 0.6,
          "lineHeight": 27.5,
          "textAlign": "left",
        },
        undefined,
        Object {
          "color": "#FFFFFF",
          "fontSize": 24,
          "fontWeight": "500",
        },
        Object {
          "color": "#FFFFFF",
        },
      ]
    }
  >
    forgot_password:btn_reset_password
  </Text>
</View>
`;

exports[`Button tests renders correctly when disabled and reversed 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderColor": "#676B7A",
      "borderRadius": 4,
      "borderWidth": 2,
      "height": 68.5,
      "justifyContent": "center",
      "opacity": 0.2,
      "width": "100%",
    }
  }
  testID="button"
>
  <Text
    allowFontScaling={false}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 24,
          "letterSpacing": 0.6,
          "lineHeight": 27.5,
          "textAlign": "left",
        },
        undefined,
        Object {
          "color": "#FFFFFF",
          "fontSize": 24,
          "fontWeight": "500",
        },
        Object {
          "color": "#676B7A",
        },
      ]
    }
  >
    forgot_password:btn_reset_password
  </Text>
</View>
`;

exports[`Button tests renders correctly with child 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#676B7A",
      "borderRadius": 4,
      "height": 68.5,
      "justifyContent": "center",
      "opacity": 1,
      "width": "100%",
    }
  }
  testID="button"
>
  Test button text
</View>
`;
